embedded-localstack/src/main/java/com/playtika/test/localstack/EmbeddedLocalStackBootstrapConfiguration.java

```diff
@@ -0,0 +1,73 @@
+package com.playtika.test.localstack;
+
+import java.util.LinkedHashMap;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.Order;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+
+import static com.playtika.test.localstack.LocalStackProperties.BEAN_NAME_EMBEDDED_LOCALSTACK;
+import static org.springframework.core.Ordered.HIGHEST_PRECEDENCE;
+
+@Slf4j
+@Configuration
+@Order(HIGHEST_PRECEDENCE)
+@ConditionalOnProperty(name = "embedded.localstack.enabled", matchIfMissing = true)
+@EnableConfigurationProperties(LocalStackProperties.class)
+public class EmbeddedLocalStackBootstrapConfiguration {
+    @ConditionalOnMissingBean(name = BEAN_NAME_EMBEDDED_LOCALSTACK)
+    @Bean(name = BEAN_NAME_EMBEDDED_LOCALSTACK, destroyMethod = "stop")
+    public ConcreteLocalStackContainer localStack(ConfigurableEnvironment environment,
+                                                  LocalStackProperties properties) {
+        log.info("Starting Localstack server. Docker image: {}", properties.dockerImage);
+
+        ConcreteLocalStackContainer localStackContainer = new ConcreteLocalStackContainer(properties.dockerImage);
+        localStackContainer.withEnv("EDGE_PORT", String.valueOf(properties.getEdgePort()))
+                           .withEnv("DEFAULT_REGION", properties.getDefaultRegion())
+                           .withEnv("HOSTNAME", properties.getHostname())
+                           .withEnv("HOSTNAME_EXTERNAL", properties.getHostnameExternal())
+                           .withEnv("USE_SSL", String.valueOf(properties.isUseSsl()));
+
+        for (String service : properties.services){
```

Comment:
can we use ```LocalStackContainer.Service``` in properties instead of converting it from String?
also it will have a validation up on missing service

Reply:
Fixed.